ui/src/views/access-point/create-ap.form-body.tsx

```diff
@@ -0,0 +1,101 @@
+import { Button, Flex, Form, Spinner, Stepper } from '@/components';
+import { AppLog } from '@/utils';
+import { useQuery } from '@apollo/client';
+import { ethers } from 'ethers';
+import { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { useAccessPointFormContext } from './create-ap.form.context';
+import { NfaPicker } from './nfa-picker';
+import { getNFADocument } from '@/graphclient';
+import { CreateAccessPoint } from './create-ap.context';
+
+export const CreateAccessPointFormBody = () => {
+  const { id } = useParams();
+  const { setNfa } = CreateAccessPoint.useContext();
+
+  const { nextStep } = Stepper.useContext();
+
+  const { nfa, billing } = CreateAccessPoint.useContext();
+  const { setArgs } = CreateAccessPoint.useTransactionContext();
+
+  const {
+    form: {
+      appName: {
+        value: [appName],
+      },
+      isValid: [isValid],
+    },
+  } = useAccessPointFormContext();
+
+  const {
+    form: { appName: appNameContext },
+  } = useAccessPointFormContext();
+
+  const {
+    data: nfaData,
+    error: nfaError,
+    loading: nfaLoading,
+  } = useQuery(getNFADocument, {
+    skip: id === undefined,
+    variables: {
+      id: ethers.utils.hexlify(Number(id)),
+    },
+  });
+
+  useEffect(() => {
+    if (nfaError) {
+      AppLog.errorToast('Error fetching NFA');
+    }
+  }, [nfaError]);
+
+  useEffect(() => {
+    if (nfaData) {
+      if (nfaData.token && id) {
+        const { name } = nfaData.token;
+        setNfa({ value: id, label: name });
+      } else {
+        AppLog.errorToast("We couldn't find the NFA you are looking for");

```

Comment:
We need to something else than just print it here. When we have the designs is good to at least clean up the search param in the URL and allow the user to select a NFA or something like that for this flow.